packages/shared/src/components/modals/bookmark/BookmarkFolderModal.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { ReactElement, useState, type FormEvent } from 'react';
+import Link from 'next/link';
+import classNames from 'classnames';
+import { usePlusSubscription, useViewSize, ViewSize } from '../../../hooks';
+import { Modal, type ModalProps } from '../common/Modal';
+import { ModalHeader } from '../common/ModalHeader';
+import { TextField } from '../../fields/TextField';
+import {
+  Typography,
+  TypographyColor,
+  TypographyTag,
+  TypographyType,
+} from '../../typography/Typography';
+import { Button, ButtonVariant } from '../../buttons/Button';
+import { FolderIcon } from '../../icons/Folder';
+import { DevPlusIcon } from '../../icons';
+import { plusUrl } from '../../../lib/constants';
+import { anchorDefaultRel } from '../../../lib/strings';
+import { LogEvent, TargetId } from '../../../lib/log';
+import { useLazyModal } from '../../../hooks/useLazyModal';
+import { IconSize } from '../../Icon';
+
+type BookmarkFolderModalProps = Omit<ModalProps, 'children'> & {
+  onSubmit: (name: string) => void;
```

Comment:
We should pass outside the current values

```
onSubmit: (folder: Record<'name'|'icon', string>) => void;
```

Reply:
Left like this to be changed when implementing the final function